engine/src/main/java/io/zeebe/engine/processing/streamprocessor/ReProcessingStateMachine.java

```diff
@@ -127,18 +128,18 @@
   private final BooleanSupplier abortCondition;
   private final Set<Long> failedEventPositions = new HashSet<>();
   // current iteration
-  private long lastSourceEventPosition;
   private long lastFollowUpEventPosition;
   private long snapshotPosition;
   private long highestRecordKey = -1L;
 
   private final Map<Long, Long> lastGeneratedKeyBySourceCommandPosition = new HashMap<>();
 
-  private ActorFuture<Long> recoveryFuture;
+  private ActorFuture<Positions> recoveryFuture;
   private LoggedEvent currentEvent;
   private TypedRecordProcessor eventProcessor;
   private ZeebeDbTransaction zeebeDbTransaction;
   private final boolean detectReprocessingInconsistency;
+  private Positions positions;
```

Comment:
I would prefer it if this were a local/stack variable in the recovery methods. As far as I can see, it seems to be only relevant during recovery. Then it is also only true during recovery.

Reply:
The complete class is only used for recovery 😊 but yes I could also use an variable. Idk whether this makes here really a difference.